File: backend/src/main/java/com/unihub/backend/model/enums/Antecedencia.java
package com.unihub.backend.model.enums;

import java.time.Duration;

public enum Antecedencia {
    NA_HORA(Duration.ZERO),
    UMA_HORA(Duration.ofHours(1)),
    DUAS_HORAS(Duration.ofHours(2)),
    TRES_HORAS(Duration.ofHours(3)),
    SEIS_HORAS(Duration.ofHours(6)),
    DOZE_HORAS(Duration.ofHours(12)),
    UM_DIA(Duration.ofHours(24)),
    DOIS_DIAS(Duration.ofDays(2)),
    TRES_DIAS(Duration.ofDays(3)),
    UMA_SEMANA(Duration.ofDays(7)),
    DUAS_SEMANAS(Duration.ofDays(14));

    private final Duration duracao;

    Antecedencia(Duration duracao) {
        this.duracao = duracao;
    }

    public Duration getDuracao() {
        return duracao;
    }

    public static Antecedencia padrao() {
        return NA_HORA;
    }
}

File: UniHub/app/src/main/java/com/example/unihub/data/model/Antecedencia.kt
package com.example.unihub.data.model

import java.time.Duration

enum class Antecedencia(val duration: Duration, val label: String) {
    NA_HORA(Duration.ZERO, "Na hora"),
    UMA_HORA(Duration.ofHours(1), "1 hora antes"),
    DUAS_HORAS(Duration.ofHours(2), "2 horas antes"),
    TRES_HORAS(Duration.ofHours(3), "3 horas antes"),
    SEIS_HORAS(Duration.ofHours(6), "6 horas antes"),
    DOZE_HORAS(Duration.ofHours(12), "12 horas antes"),
    UM_DIA(Duration.ofHours(24), "24 horas antes"),
    DOIS_DIAS(Duration.ofDays(2), "2 dias antes"),
    TRES_DIAS(Duration.ofDays(3), "3 dias antes"),
    UMA_SEMANA(Duration.ofDays(7), "1 semana antes"),
    DUAS_SEMANAS(Duration.ofDays(14), "2 semanas antes");

    companion object {
        val padrao = NA_HORA
        val todas = listOf(
            NA_HORA,
            UMA_HORA,
            DUAS_HORAS,
            TRES_HORAS,
            SEIS_HORAS,
            DOZE_HORAS,
            UM_DIA,
            DOIS_DIAS,
            TRES_DIAS,
            UMA_SEMANA,
            DUAS_SEMANAS
        )
    }
}